README.md

```diff
@@ -266,6 +266,29 @@ Declarative Pipeline support requires Jenkins 2.66+
 
 Example at [examples/declarative.groovy](examples/declarative.groovy)
 
+### Referring to PodTemplates
+
+You can refer to a named PodTemplate in the Jenkins configuration via the **inheritFrom** property as follows:
+
+```groovy
+pipeline {
+  agent {
+    kubernetes {
+      label "my-build-pod"
+      inheritFrom "maven"
+    }
+  }
+  ...
+}
+``` 
+
+The meaning of the values is as follows:
+
+* **label** is used to add a kubernetes label to the build pods created by the pipeline jobs
```

Comment:
So I guess I'm now confused why there are `label` and `inheritFrom` properties on the `KubernetesDeclarativeAgent` - I figured `label` was so that you could override the label used?

Reply:
I think one thing that confuses me is terminology. What I understand with `kubernetes labels` is the labels present in the metadata of the kubernetes resource. For the labels used in order to select nodes I usually go with `jenkins labels`.

The other thing that confuses me, is that `label` and `inheritFrom` seem to be usable both in and outside of the `podTemplate`.

From what I understand there used to be no notion of `podTemplate` under `kubernetes` and there were just raw key/value pairs (subset of the actual `podTemplate`).

I really like the idea of being able to have full control of the podTemplate from declerative pipelines. So +1 on this one. But we need to make it a bit more clear on how things work. To be fair this has to do more with how it already works, and less with what this pr brings in.